common/src/main/java/com/microsoft/identity/common/crypto/AndroidWrappedKeyLoader.java

```diff
@@ -0,0 +1,335 @@
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.crypto;
+
+import android.annotation.TargetApi;
+import android.content.Context;
+import android.os.Build;
+import android.security.KeyPairGeneratorSpec;
+
+import androidx.annotation.RequiresApi;
+
+import com.microsoft.identity.common.adal.internal.AuthenticationConstants;
+import com.microsoft.identity.common.internal.util.AndroidKeyStoreUtil;
+import com.microsoft.identity.common.internal.util.FileUtil;
+import com.microsoft.identity.common.java.crypto.key.AES256KeyLoader;
+import com.microsoft.identity.common.java.crypto.key.KeyUtil;
+import com.microsoft.identity.common.java.exception.ClientException;
+import com.microsoft.identity.common.java.telemetry.ITelemetryCallback;
+import com.microsoft.identity.common.logging.Logger;
+
+import java.math.BigInteger;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.spec.AlgorithmParameterSpec;
+import java.util.Calendar;
+import java.util.Locale;
+
+import javax.crypto.SecretKey;
+import javax.security.auth.x500.X500Principal;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import lombok.NonNull;
+
+/**
+ * This class doesn't really use the KeyStore-generated key directly.
+ * <p>
+ * Instead, the actual key that we use to encrypt/decrypt data is 'wrapped/encrypted' with the keystore key
+ * before it get saved to the file.
+ */
+@TargetApi(Build.VERSION_CODES.JELLY_BEAN_MR2)
+public class AndroidWrappedKeyLoader extends AES256KeyLoader {
+    private static final String TAG = AndroidWrappedKeyLoader.class.getSimpleName() + "#";
+
+    /**
+     * Alias for this type of key.
+     */
+    /* package */ static final String KEYSTORE_KEY_ALIAS = "KEYSTORE_KEY";
+
+    /**
+     * Algorithm for key wrapping.
+     */
+    private static final String WRAP_ALGORITHM = "RSA/ECB/PKCS1Padding";
+
+    /**
+     * Algorithm for the wrapping key itself.
+     */
+    private static final String WRAP_KEY_ALGORITHM = "RSA";
```

Comment:
Can we take all algorithm Strings used throughout this PR, and put them in an Enum?

Reply:
I believe there is already a `SigningAlgorithm` enum in this PR

Reply:
Those are for something else. I think creating enum for this is an overkill.